microsoft/testsuites/mdatp/mdatp_tools.py

```diff
@@ -0,0 +1,93 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+import json
+from typing import Any
+
+from lisa.tools import Chmod
+from lisa.base_tools import Wget
+from lisa.executable import Tool
+
+class Mdatp(Tool):
+    @property
+    def command(self) -> str:
+        return "mdatp"
+
+    @property
+    def can_install(self) -> bool:
+        return True
+
+    def get_mde_installer(self) -> bool:
+        if not hasattr(self, "mde_installer"):
+            wget = self.node.tools[Wget]
+
+            download_path = wget.get(
+                url="https://raw.githubusercontent.com/microsoft/mdatp-xplat/"
+                "master/linux/installation/mde_installer.sh",
+                filename="mde_installer.sh",
+            )
+            self.mde_installer = download_path
+            self.node.tools[Chmod].update_folder(self.mde_installer, "777", sudo=True)
+        return True
+
+    def _install(self) -> bool:
+        if not self.get_mde_installer():
+            self._log.error(
+                "Unable to download mde_installer.sh script. MDE can't be installed"
+            )
+
+        self._log.info("Installing MDE")
+        result1 = self.node.execute(
+            f"{self.mde_installer} --install", shell=True, sudo=True
+        )
+        self._log.info(result1)
+
+        return self._check_exists()
+
+    def onboard(self, onboarding_script_sas_uri: str) -> bool:
+        if not self._check_exists():
```

Comment:
It doesn't need to check, when you have an instance of the tool, it means the check exists already passed.